src/characters/bob.ts

```diff
@@ -0,0 +1,166 @@
+import { PublicKey, Signer } from 'umbral-pre';
+
+import { verifySignature } from '../crypto/api';
+import {
+  SIGNATURE_HEADER_BYTES_LENGTH,
+  SIGNATURE_HEADER_HEX,
+  SIGNATURE_LENGTH,
+} from '../crypto/constants';
+import { NucypherKeyring } from '../crypto/keyring';
+import { DecryptingPower, SigningPower } from '../crypto/powers';
+import { MessageKit, PolicyMessageKit } from '../kits/message';
+import {
+  EncryptedTreasureMap,
+  PublishedTreasureMap,
+} from '../policies/collections';
+import { Configuration } from '../types';
+import { bytesEqual, split, toHexString } from '../utils';
+
+import { Porter } from './porter';
+
+export class Bob {
+  private readonly treasureMaps: Record<string, PublishedTreasureMap>;
+  private readonly config: Configuration;
+  private readonly porter: Porter;
+  private readonly signingPower: SigningPower;
+  private readonly decryptingPower: DecryptingPower;
+
+  constructor(
+    config: Configuration,
+    signingPower: SigningPower,
+    decryptingPower: DecryptingPower
+  ) {
+    this.config = config;
+    this.porter = new Porter(config.porterUri);
+    this.signingPower = signingPower;
+    this.decryptingPower = decryptingPower;
+    this.treasureMaps = {};
+  }
+
+  public get decryptingKey(): PublicKey {
+    return this.decryptingPower.publicKey;
+  }
+
+  public get verifyingKey(): PublicKey {
+    return this.signingPower.publicKey;
+  }
+
+  public get signer(): Signer {
+    return this.signingPower.signer;
+  }
+
+  public static fromPublicKeys(
```

Comment:
Perhaps we do not have to repeat the current design here and can separate local and remote characters instead? Now, for example, a remote Bob still has a Porter instantiated inside it.